nanomesh/generator.py

```diff
@@ -0,0 +1,61 @@
+
+import logging
+import numpy as np
+import math
+
+# Pores in X direction are at
+# y,z= (0,0) (a,0) (0,c) (a,c) and (a/2,c/2)
+
+# Pores in Z direction are at
+# x,y = (0,a/4) (c,a/4) and (c/2,3a/4)
+
+
+# Generator for a theoretical structure
+class Generator(object):
+
+    def __init__(self, a, c, r):
+        # Long crystal axis, usually a = sqrt(c)
+        self.a = a
+        self.c = c
+        # Pore radius
+        self.r = r
+
+    def generate(self, sizes, resolution, transform=None):
```

Comment:
Could you write doc strings (or type hints) so that we know what should be the types of the different arguments ? (I also don't do it all the time so I feel a bit hypocritical here :) )

Reply:
but especially transform here I wouldn't know what it is supposed to be. I guess a np array but we could also have a callable

Reply:
@NicoRenaud I have added documentation and most type hints (those for the ndarrays are missing). I hope it is now much clearer.